messaginginapp/src/main/java/io/customer/messaginginapp/ModuleMessagingInApp.kt

```diff
@@ -34,19 +34,31 @@ class ModuleMessagingInApp internal constructor(
 
     private val gistProvider by lazy { diGraph.gistProvider }
 
+    private val logger by lazy { diGraph.logger }
+
     override fun initialize() {
         initializeGist(organizationId)
         setupHooks()
         setupGistCallbacks()
     }
 
     private fun setupGistCallbacks() {
-        gistProvider.subscribeToEvents { deliveryID ->
-            trackRepository.trackInAppMetric(
-                deliveryID = deliveryID,
-                event = MetricEvent.opened
-            )
-        }
+        gistProvider.subscribeToEvents(
+            onMessageShown = { deliveryID ->
+                trackRepository.trackInAppMetric(
+                    deliveryID = deliveryID,
+                    event = MetricEvent.opened
+                )
+            },
+            onAction = { deliveryID: String, _: String, _: String ->
+                trackRepository.trackInAppMetric(
+                    deliveryID = deliveryID,
+                    event = MetricEvent.clicked
+                )
+            }, onError = { error ->
+                logger.error("in-app message error occurred $error")
```

Comment:
I wish that we could include some reference to the failed message in the error message. That way when there is an error, we know what message the error happened for. 

Do you think there is a way? 

```suggestion
                logger.error("in-app message error occurred $error. message: $message")
```

Reply:
Ah, I see. I think the variable `error` was what got me confused. I thought that it was an object. Perhaps a variable name such as `errorMessage` or `errorDescription` would be better?